### PR TITLE
add examples of deployment object

### PR DESCRIPTION
**Pull request by DjangoPeng**
Add 5 examples of deployment object, including:
1. Create deployment object with client-python API
2. Create deployment
3. Update deployment
4. Roll back deployment
5. Cascading delete deployment

@mbohlool Please check the PR.  Refer this [issue](https://github.com/kubernetes-incubator/client-python/issues/320)

### CODE REVIEW

**Comment by DjangoPeng** [comment]
Ping @mbohlool .I have updated the code style for PEP8. Now, it's ready to be reviewed.

File: examples/deployment_examples.py [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+        body=deployment,
+        namespace="default")
+    print("Deployment created. status='%s'" % str(api_response.status))
+
+
+def update_deployment(api_instance, deployment):
+    # Update container image
+    deployment.container.image = "nginx:1.9.1"
+    # Update the deployment
+    api_response = api_instance.replace_namespaced_deployment(
```
Comment (mbohlool) [inline examples/deployment_examples.py]: I suggest you use `patch_namespaced_deployment` here as it is the recommended way to update an API object. Replace can have unintentional consequences.

Comment (DjangoPeng) [reply examples/deployment_examples.py]: Good suggestion.

**Comment by DjangoPeng** [comment]
@mbohlool Updated. Plz review.

**Comment by mbohlool** [comment]
Have you actually tried this code? specially the patch that you just changed? I just want to make sure it is working for you before merging it into master.

**Comment by mbohlool** [comment]
@DjangoPeng I tried these examples and they gave me syntax errors. That could be because of the new client generation. Can you please take a look at them and make sure that they can be run?

**Comment by DjangoPeng** [comment]
@mbohlool I have fixed the error and tested all examples by myself. Please review and check.

**Comment by mbohlool** [comment]
@DjangoPeng The branch is out of date. The new client has some differences (like you need to provide required fields on model creation, etc.). Please rebase to master and run the examples again. An example failure I get:

```
ValueError: Invalid value for `template`, must not be `None`
```

**Comment by DjangoPeng** [comment]
@mbohlool Ok, I'll test it today .

**Comment by DjangoPeng** [comment]
I run all examples on Kubernetes 1.7 with client-python 3.0, and I attached the stdout as below. Should I try it on Kubernetes 1.8, or what?
```
$ python deployment_examples.py
Deployment created. status='{'available_replicas': None,
 'collision_count': None,
 'conditions': None,
 'observed_generation': None,
 'ready_replicas': None,
 'replicas': None,
 'unavailable_replicas': None,
 'updated_replicas': None}'
Deployment updated. status='{'available_replicas': None,
 'collision_count': None,
 'conditions': [{'last_transition_time': datetime.datetime(2017, 10, 13, 1, 30, 1, tzinfo=tzutc()),
                 'last_update_time': datetime.datetime(2017, 10, 13, 1, 30, 1, tzinfo=tzutc()),
                 'message': 'Deployment does not have minimum availability.',
                 'reason': 'MinimumReplicasUnavailable',
                 'status': 'False',
                 'type': 'Available'}],
 'observed_generation': 1,
 'ready_replicas': None,
 'replicas': None,
 'unavailable_replicas': 3,
 'updated_replicas': None}'
Deployment deleted. status='{u'observedGeneration': 1, u'unavailableReplicas': 3, u'conditions': [{u'status': u'False', u'lastUpdateTime': u'2017-10-13T01:30:01Z', u'lastTransitionTime': u'2017-10-13T01:30:01Z', u'reason': u'MinimumReplicasUnavailable', u'message': u'Deployment does not have minimum availability.', u'type': u'Available'}]}'
```

**Comment by mbohlool** [comment]
Use master. You can goto root folder of repo and call `python -m examples.deploment_examples`. just make sure you rebase to the latest code.

**Comment by DjangoPeng** [comment]
@mbohlool Now, it works on latest master branch. I delete the rollback example because there is a error that I'm debugging and identifying. When I solve it, I'll make a new PR to append the rollback example.

**Comment by DjangoPeng** [comment]
@mbohlool Please review the PR. I think it's ready to be merged.

**Comment by mbohlool** [comment]
@DjangoPeng Thanks. Great example.

**Comment by DjangoPeng** [comment]
I have updated the description of the PR. Welcome anyone to append the rollback example of Deployment.

**Pull request merged**